lib/chai/core/assertions.js

```diff
@@ -537,8 +537,8 @@ module.exports = function (chai, _) {
         itemsCount = val.length;
         break;
       case 'function':
-        var name = val.name ? ' ' + val.name : '';
-        throw new TypeError('.empty was passed a function' + name);
+        var msg = '.empty was passed a function ' + _.getName(val);
+        throw new TypeError(msg.trim());
```

Comment:
Yep, that is exactly why I used `trim` here. However, because function `name` property is `configurable` in ES6, it might contain trailing spaces.